### ISSUE TITLE
Add the "-overwrite" flag to the "clip raster by mask" GDAL tool

### ISSUE THREAD

**Issue by glw** [qgis/QGIS#31380]
**Describe the bug**

When atempting to overwrite a previous raster (in this case) using clip raster by mask, it is not over written even when QGIS prompts you to overwrite.

**How to Reproduce**
Raster > Extraction > Clip Raster by Mask layer
Save the file to a directory
Go through the same process, save as same file name, and agree to overwrite the file.
Run clipping process again. 
See error message below.

**QGIS and OS versions**
QGIS: 3.4.10-Madeira
OS: Ubuntu Xenial
GDAL/OGR: 2.2.2

**Additional context**

Error Message:

```
Clip Raster by Mask Layer

ERROR 1: Output dataset /filename_clipped.tif exists,

but some command line options were provided indicating a new dataset

should be created. Please delete existing dataset and run again.
```
I think here when running the tool the ```-overwrite```  needs to be specified?

**Comment by gioman** [comment]
Yes, it seems we miss the "-overwrite" flag in this GDAL command.

**Comment by alexbruy** [comment]
This already implemented in 3.8/master
